src/NetUtils.cpp

```diff
@@ -18,39 +18,60 @@ std::string NetUtils::getSelfIpAddress(void)
 {
 	std::string ipAddress;
 
-	struct ifaddrs *ifaddr, *ifa;
-	int family, s;
-	char host[NI_MAXHOST];
-
-	if (getifaddrs(&ifaddr) == -1) {
-	   perror("getifaddrs");
-	   return std::string("");
-	}
-
-	/* Walk through linked list, maintaining head pointer so we
-	  can free list later */
-
-	ifa = ifaddr;
-	if (ifa == NULL || ifa->ifa_addr == NULL)
-		return std::string("");
-
-	family = ifa->ifa_addr->sa_family;
-
-	if (family == AF_INET || family == AF_INET6) {
-	   s = getnameinfo(ifa->ifa_addr,
-			   (family == AF_INET) ? sizeof(struct sockaddr_in) :
-									 sizeof(struct sockaddr_in6),
-			   host, NI_MAXHOST,
-			   NULL, 0, NI_NUMERICHOST);
-	   if (s != 0) {
-		   printf("getnameinfo() failed: %s\n", gai_strerror(s));
-		   return std::string("");
-	   }
-	}
-
-	freeifaddrs(ifaddr);
-
-
+    struct ifaddrs *ifaddr, *ifa;
+    int family, s, n;
+    char host[NI_MAXHOST];
+
+    if (getifaddrs(&ifaddr) == -1) {
+        perror("getifaddrs");
+        return std::string("");
+    }
+
+    /* Walk through linked list, maintaining head pointer so we
+       can free list later */
+
+    for (ifa = ifaddr, n = 0; ifa != NULL; ifa = ifa->ifa_next, n++) {
+        if (ifa->ifa_addr == NULL)
+            continue;
+
+        family = ifa->ifa_addr->sa_family;
+
+        /* Display interface name and family (including symbolic
+           form of the latter for the common families) */
+
+        printf("%-8s %s (%d)\n",
+               ifa->ifa_name,
+               (family == AF_PACKET) ? "AF_PACKET" :
+               (family == AF_INET) ? "AF_INET" :
+               (family == AF_INET6) ? "AF_INET6" : "???",
+               family);
+
+        /* For an AF_INET* interface address, display the address */
+
+        if (family == AF_INET || family == AF_INET6) {
+            s = getnameinfo(ifa->ifa_addr,
+                    (family == AF_INET) ? sizeof(struct sockaddr_in) :
+                                          sizeof(struct sockaddr_in6),
+                    host, NI_MAXHOST,
+                    NULL, 0, NI_NUMERICHOST);
+            if (s != 0) {
+                printf("getnameinfo() failed: %s\n", gai_strerror(s));
+                return std::string("");
+            }
+
+            printf("\t\taddress: <%s>\n", host);
+
+        } else if (family == AF_PACKET && ifa->ifa_data != NULL) {
+            struct rtnl_link_stats *stats = (struct rtnl_link_stats *)ifa->ifa_data;
+
+            printf("\t\ttx_packets = %10u; rx_packets = %10u\n"
```

Comment:
i think we should have one convention, cout or printf, most code's already using cout so i suggest to change